core/src/components/datetime/datetime.tsx

```diff
@@ -1278,6 +1294,21 @@ export class Datetime implements ComponentInterface {
             const referenceParts = { month, day, year };
             const { isActive, isToday, ariaLabel, ariaSelected, disabled } = getCalendarDayState(this.locale, referenceParts, this.activePartsClone, this.todayParts, this.minParts, this.maxParts, this.parsedDayValues);
 
+            let isCalDayDisabled = isCalMonthDisabled || disabled;
+
+            if (!isCalDayDisabled && this.isDateEnabled !== undefined) {
```

Comment:
Non-blocking, but doing `const { isDateEnabled } = this` lets you avoid a bit of overhead on line 1306 when accessing `isDateEnabled` again